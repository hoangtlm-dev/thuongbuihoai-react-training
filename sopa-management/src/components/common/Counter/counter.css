.counter {
  border-radius: var(--rounded-secondary);
  border: 1px solid var(--border-primary);
  height: var(--h-normal);
  width: var(--w-normal);
}

.counter-minus {
  width: 40%;
  display: block;
  height: 44px;
  float: left;
  position: relative;
}

.counter-plus {
  position: absolute;
  top: 38px;
  left: 130px;
}

.counter-minus:before,
.counter-plus:before,
.counter-plus:after {
  content: '';
  position: absolute;
  background-color: var(--text-tertiary);
  height: 2px;
  width: 10px;
  top: 50%;
  left: 50%;
  margin-top: -1px;
  margin-left: -5px;
}

.counter-plus:after {
  transform: rotate(90deg);
}

.counter-input {
  width: 20%;
  font-size: var(--text-base);
  font-weight: var(--font-semi-bold);
  height: 42px;
  border: none;
  text-align: center;
}

.counter-input::-webkit-outer-spin-button,
.counter-input::-webkit-inner-spin-button {
  -webkit-appearance: none;
  margin: 0;
}
