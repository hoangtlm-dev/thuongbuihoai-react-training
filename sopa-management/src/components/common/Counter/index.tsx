import { memo } from 'react';

// Components
import Button, { ButtonType } from '@common/Button';
import Input from '../Input';

// Styles
import './counter.css';

interface CounterProps {
  value: number;
  onIncrement: () => void;
  onDecrement: () => void;
}

const Counter = ({
  value,
  onIncrement,
  onDecrement
}: CounterProps) => (
  <div data-testid='counter' className='counter'>
    <Button
      ariaLabel='Decrement'
      type={ButtonType.btnIconPrimary}
      className='counter-minus'
      onClick={onDecrement}
    />
    <Input type='number' classNameInput='counter-input' value={value} />
    <Button
      ariaLabel='Increment'
      type={ButtonType.btnIconPrimary}
      className='counter-plus'
      onClick={onIncrement}
    />
  </div>
);

export default memo(Counter);
