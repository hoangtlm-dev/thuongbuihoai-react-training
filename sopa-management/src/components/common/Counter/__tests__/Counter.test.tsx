// Libs
import '@testing-library/jest-dom';
import { render } from '@testing-library/react';

// Components
import Counter from '..';

describe('Counter component', () => {
  test('should render Counter component', () => {
    const { getByTestId } = render(<Counter />);
    expect(getByTestId('counter')).toBeInTheDocument();
  });

  test('matches snapshot', () => {
    const { asFragment } = render(<Counter />);
    expect(asFragment()).toMatchSnapshot();
  });
});
