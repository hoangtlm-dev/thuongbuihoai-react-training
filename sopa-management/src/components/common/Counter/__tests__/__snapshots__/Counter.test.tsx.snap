// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="counter"
    data-testid="counter"
  >
    <button
      class="counter-minus btn-icon-primary"
      data-testid="button"
    />
    <input
      class="counter-input"
      type="number"
      value="0"
    />
    <button
      class="counter-plus btn-icon-primary"
      data-testid="button"
    />
  </div>
</DocumentFragment>
`;
